core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -1415,6 +1415,79 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
       super.fetchBlockSync(host, port, execId, blockId)
     }
   }
+
+  def testGetOrElseUpdateForLargeBlock(storageLevel: StorageLevel) {
```

Comment:
Have you measured how long these tests take? I've seen this tried before in other changes related to 2g limits, and this kind of test was always ridiculously slow.

You can avoid this kind of test by making the chunk size configurable, e.g. in this line you're adding above:

    val chunkSize = math.min(remaining, Int.MaxValue)

Then your test can run fast and not use a lot of memory. You just need to add extra checks that the data is being chunked properly, instead of relying on the JVM not throwing errors at you.